### PR TITLE
Update .asf.yaml to publish docs to datafusion.apache.org

### LINKED ISSUE #10151: Create Project Website (datafusion.apache,org)
### Is your feature request related to a problem or challenge?

We need to create a website for the project, to be hosted at datafusion.apache.org

See https://infra.apache.org/project-site.html for more information 

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by phillipleblanc**
## Which issue does this PR close?

Closes #10151

## What changes are included in this PR?

Updates `.asf.yaml` to point to the new top-level website, and adds the required links to the ASF as required by https://www.apache.org/foundation/marks/pmcs#navigation

I mirrored what https://arrow.apache.org does and grouped these links into an `ASF Links` section

According to https://github.com/apache/infrastructure-asfyaml/blob/main/README.md#publish - removing the `subdir` config in `publish` will do what we want.

## Are these changes tested?

Here is a screenshot of what it looks like locally:
Screenshot 2024-04-23 at 11 40 35 AM

## Are there any user-facing changes?

N/A

### CODE REVIEW

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @phillipleblanc. I think it would be good to wait for @alamb to also review before merging.

**Comment by phillipleblanc** [comment]
> LGTM. Thanks @phillipleblanc. I think it would be good to wait for @alamb to also review before merging.

Yeah - there are some links to arrow.apache.org/datafusion still leftover in the repo as well - but I also think we should wait on those changes until we confirm this actually publishes to the right place.

Also I wasn't able to figure out how to add the redirect from arrow.apache.org/datafusion to datafusion.apache.org. @alamb did you have an idea on how to do that?

File: .asf.yaml
Comment (viirya) [inline .asf.yaml:30]: It should be `https://datafusion.apache.org/datafusion/`.

Comment (viirya) [reply .asf.yaml:30]: `https://datafusion.apache.org/` will give you:

Screenshot 2024-04-22 at 8 19 35 PM

Comment (phillipleblanc) [reply .asf.yaml:30]: After this change it should start publishing to https://datafusion.apache.org without needing the subdirectory 🤞

**Review by viirya — APPROVED** [review]

**Comment by alamb** [comment]
> Also I wasn't able to figure out how to add the redirect from arrow.apache.org/datafusion to datafusion.apache.org. @alamb did you have an idea on how to do that?

The only thing I could think of is creating explicit redirect pages for each .html page in  https://github.com/apache/arrow-site

I agree having redirects will be important (and I m am not sure what will be on https://arrow.apache.org/datafusion/ afterwards).

Thus what I think we should do is to merge this PR, see its effects and I'll file a follow on ticket for the next tasks

**Comment by alamb** [comment]
Thank you for your work / help and reviews @phillipleblanc

**Comment by alamb** [comment]
Filed https://github.com/apache/datafusion/issues/10194 to track redirects

**Comment by alamb** [comment]
It appears to have worked!

https://datafusion.apache.org/ now shows:

Screenshot 2024-04-23 at 6 43 58 AM

**Comment by alamb** [comment]
It appears that https://arrow.apache.org/datafusion/ has an older snapshot of the site (as in it was not updated as well):

Screenshot 2024-04-23 at 6 45 03 AM

**Pull request merged**
